jellyfin-media-player: 1.12.0 -> ???

picnoir opened the pull request:
WIP, waiting for upstream to cut a new release.

Creating this patch early on to benefit from the newly acquired upstream qt6 support.

I got rid of the web patch. It seems to be outdated.

TODO: review all the libs this derivation is depending on. I have the feeling we probably can get of some.

In the meantime, feel free to test that and see if anything's broken.

Fixes https://github.com/NixOS/nixpkgs/issues/437865


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

picnoir committed:
jellyfin-media-player: 1.12.0 -> ???

WIP, waiting for upstream to cut a new release.

Creating this patch early on to benefit from the newly acquired
upstream qt6 support.

I got rid of the web patch. It seems to be outdated.

TODO: review all the libs this derivation is depending on. I have the
feeling we probably can get of some.

mio-19 commented:
mpvqt on nixpkgs is marked broken on darwin. it recently added mpvqt submodule https://github.com/jellyfin/jellyfin-media-player/blob/master/.gitmodules when I added fetchsubmodules the latest master branch builds on darwin; however it crashed during launching showing iconv related errors

mio-19 commented:
> Do we know what's broken exactly? Could somebody paste failed build log somewhere?

it is not actually broken but darwin isn't marked as supported. We just need to add darwin to meta.platforms of mpvqt and a qtcmake derivation

This pr works with the Randy patch on Randy's darwin device but not my darwin device. There is other error on my device:

```
Detected locale "C" with character encoding "US-ASCII", which is not UTF-8. Qt depends on a UTF-8 locale, and has switched to "UTF-8" instead. If this causes problems, reconfigure your locale. See the locale(1) manual for more information. DisplayManager : unable to retrieve current video mode Slot for host command missing, or has incorrect signature! [49587:259:1127/153935.932187:ERROR:service_utils.cc(339)] Skia Graphite backend = "" not found - falling back to Ganesh! [49587:259:1127/153935.936711:ERROR:service_utils.cc(339)] Skia Graphite backend = "" not found - falling back to Ganesh! libc++abi: terminating due to uncaught exception of type FatalException: std::exception Received signal 6  [0x00011d6444b0]  [0x00011d62f9f0]  [0x00011d6443c8]  [0x0001992216a4]  [0x0001991e7848]  [0x0001990f09e4]  [0x00019919d384]  [0x00019918bcd8]  [0x000198e10de4]  [0x00019919c698]  [0x00019919fc30]  [0x00019919fbd8]  [0x00010098afb0]  [0x000104d98508]  [0x0001033f97a8]  [0x0001033f9cc0]  [0x0001009876b0]  [0x000198e46b98] [end of stack trace] zsh: abort      nix run .#jellyfin-media-player --impure
```

picnoir committed:
mpvqt: add darwin platforms to supported platforms

Co-authored-by: Mio <mio@mio19.uk>

reckenrode committed:
jellyfin-media-player: fix darwin build

picnoir commented:
Added darwin to the supported platforms of `mpvqt` and added @reckenrode patch. Again, operating blind, I don't have a darwin system to test on.

I assume we'd need the debug symbols and some sort of darwin debugger to get some sort of stacktrace to understand what's throwing the exception?

miangraham commented:
Here's some more input seen while running under lldb. No exceptions or fatal errors observed, just the forever spinner:

```console
λ lldb ./result/bin/jellyfinmediaplayer
(lldb) target create "./result/bin/jellyfinmediaplayer"
Current executable set to '/Users/ian/git/nixpkgs/result/bin/jellyfinmediaplayer' (arm64).
(lldb) settings set target.process.stop-on-exec false
(lldb) r
Process 26773 launched: '/Users/ian/git/nixpkgs/result/bin/jellyfinmediaplayer' (arm64)
2025-11-28 09:48:08.628726+0900 .Jellyfin Media Player-wrapped[26773:25565651] Detected locale "C" with character encoding "US-ASCII", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
2025-11-28 09:48:08.768465+0900 .Jellyfin Media Player-wrapped[26773:25565651] [mpv] AppHub initialised
2025-11-28 09:48:08.768477+0900 .Jellyfin Media Player-wrapped[26773:25565651] [mpv] Initialising Input
2025-11-28 09:48:08.772370+0900 .Jellyfin Media Player-wrapped[26773:25565651] [mpv] AppHub functionality initialised
2025-11-28 09:48:08.773476+0900 .Jellyfin Media Player-wrapped[26773:25565651] [mpv] Initialising Input
Slot for host command missing, or has incorrect signature!
[26773:259:1128/094808.815905:ERROR:service_utils.cc(339)] Skia Graphite backend = "" not found - falling back to Ganesh!
[26773:259:1128/094808.816101:ERROR:service_utils.cc(339)] Skia Graphite backend = "" not found - falling back to Ganesh!
2025-11-28 09:48:09.119044+0900 .Jellyfin Media Player-wrapped[26773:25565765] [mpv] Initialising Input
2025-11-28 09:48:09.120110+0900 .Jellyfin Media Player-wrapped[26773:25565765] [mpv] AppHub functionality initialised
2025-11-28 09:48:09.174090+0900 .Jellyfin Media Player-wrapped[26773:25565766] [plugin] AddInstanceForFactory: No factory registered for id <CFUUID 0x600000734080> F8BB1C28-BAE8-11D6-9C31-00039315CD46
Detected locale "C" with character encoding "US-ASCII", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
```

reckenrode commented:
I also just get a spinner in the window. There are more logs in `~/Library/Logs/Jellyfin Media Player`. I see this repeating when I try to connect:

```
2025-11-27 20:58:08.392 [info] unknown @ 0 - JS: qrc:///web-client/extension/find-webclient.js:7 Checking connectivity to: https://<hostname>/jellyfin
2025-11-27 20:58:08.631 [warning] unknown @ 0 - resolveUrl: error: "SSL handshake failed: The issuer certificate of a locally looked up certificate could not be found"
2025-11-27 20:58:08.651 [warning] unknown @ 0 - checkServerConnectivity: error: "SSL handshake failed: The issuer certificate of a locally looked up certificate could not be found"
```

Which is weird because the certificate is a valid certificate provisioned from Let’s Encrypt using the `security.acme` module.

PartlyAwesome commented:
To add on to @numkem 's post, I am experiencing the same under nixos unstable.
[jmp-git-sigabrt.txt](https://github.com/user-attachments/files/23861457/jmp-git-sigabrt.txt)

kevincox commented:
This is probably obsoleted by https://github.com/NixOS/nixpkgs/pull/469223?

picnoir commented:
Yup
